packages/common/src/hooks/useOwnedTokens.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+
+import {
+  useCurrentUserId,
+  useUserCoins,
+  useQueryContext,
+  UserCoin
+} from '~/api'
+import { useFeatureFlag } from '~/hooks'
+import { FeatureFlags } from '~/services'
+import type { TokenInfo } from '~/store'
+import { filterUserCoins } from '~/utils'
+
+/**
+ * Hook to filter tokens based on user ownership and positive balance
+ * Respects the ARTIST_COINS feature flag - when disabled, only shows AUDIO tokens
+ * Returns both owned tokens and all available tokens for different use cases
+ */
+export const useOwnedTokens = (allTokens: TokenInfo[]) => {
```

Comment:
what's going on with `allTokens`? it is an arg but then gets returned with no changes?
can we just select `allTokens` from some other hook inside this hook?
and maybe don't need to return it?